library/core/class.smarty.php

```diff
@@ -12,7 +12,7 @@
 /**
  * Vanilla implementation of Smarty templating engine.
  */
-class Gdn_Smarty {
+class Gdn_Smarty implements \Vanilla\Contracts\Web\LegacyViewHandlerInterface {
```

Comment:
Now I see why. 
You took `render()` from here with `controller` argument

But I would say it seem that analogue of `twig->render`  is `smarty->display`.

Reply:
Except the controller does not call display, it call render.